src/Tests/AzureClientEntryPointTests.cs

```diff
@@ -81,6 +82,94 @@ public async Task FromSnippet()
             Assert.IsNotNull(job);
         }
 
+        [TestMethod]
+        public async Task QIRSubmission()
+        {
+            var entryPointGenerator = Init("Workspace", new string[] { SNIPPETS.HelloQ });
+            var entryPoint = entryPointGenerator.Generate("HelloQ", null);
+
+            Assert.IsNotNull(entryPoint);
+            var job = await entryPoint.SubmitAsync(
+                new MockQIRSubmitter(new List<Argument>()),
+                new AzureSubmissionContext());
+            Assert.IsNotNull(job);
+        }
+
+        [TestMethod]
+        public async Task ValidParameterTypes()
+        {
+            var entryPointGenerator = Init("Workspace", new string[] { SNIPPETS.ValidParameterTypes });
+            var entryPoint = entryPointGenerator.Generate("ValidParameterTypes", null);
+
+            Assert.IsNotNull(entryPoint);
+
+            var validArguments = new List<(Argument, string)>()
+            {
```

Comment:
You may be able to use dictionary initializer syntax here to make this test case a bit simpler; `validArguments.Select(x => X.Item1)` below can then be replaced by `validArguments.Keys`.

Reply:
done